MyCryptoHQ/MyCrypto
# Node Refactor

Closes #1353. Closes #1455.

### Description

~This is the first of potentially a few node refactor PRs that will make it easier to work with nodes & shepherd. This one cleans up some of the leftovers from the shepherd conversion, and reduces the number of places where we have to define nodes. It also reduces some of the reliance on implicit config naming.~

Additional PRs have been merged, see #1660, #1654, #1652 & #1640 for those. This PR contains them all, in addition to the below changes.

### Changes

* Consolidate static node definitions from being split across `libs/nodes/index.ts`, `reducers/config/nodes/staticNodes.ts`, and `types/node.d.ts` to all being located in `libs/nodes/configs.ts`
* Create `makeNodeName` and `makeAutoNodeName` functions to reduce chance of making inconsistent node names
    * This required renaming a few nodes, only for the alternative networks & testnets
* Shepherd auto configs are now automatically created for each network, instead of hard coded
    * Shepherd node configs also have an `isAuto` property, for future use
* Remove unused `lib` property from `StaticNodeConfig` and `CustomNodeConfig`

### Steps to Test

1. Connect to various nodes, ensure they work
2. Connect using web3, ensure it still works
